src/ppspp_datagram.erl

```diff
@@ -33,14 +33,25 @@
          unpack/3,
          pack/1]).
 
--opaque endpoint() :: {endpoint, list( endpoint_option())}.
--opaque endpoint_option() :: {ip, inet:ip_address()}
-| {socket, inet:socket()}
-| {port, inet:port_number()}
-| {transport, udp}
-| {uri, string()}
-| ppspp_channel:channel().
--opaque datagram() :: {datagram, list(endpoint() | ppspp_message:messages())}.
+%% -opaque endpoint()        :: {endpoint, list( endpoint_option())}.
+%% -opaque endpoint_option() :: {ip, inet:ip_address()}
+%%                            | {socket, inet:socket()}
+%%                            | {port, inet:port_number()}
+%%                            | {transport, udp}
+%%                            | {uri, string()}
+%%                            | ppspp_channel:channel().
+
+-opaque endpoint()        :: #{}.
+-opaque endpoint_option() :: #{ip => inet:ip_address()}
+                           | #{socket => inet:socket()}
+                           | #{port => inet:port_number()}
+                           | #{transport => udp}
+                           | #{uri => string()}
+                           | ppspp_channel:channel().
+
+
+-opaque datagram()        :: {datagram, map(endpoint() | ppspp_message:messages())}.
```

Comment:
Line 53 is too long: -opaque datagram()        :: {datagram, map(endpoint() | ppspp_message:messages())}..